pandas/core/frame.py

```diff
@@ -3104,7 +3104,7 @@ def _setitem_frame(self, key, value):
 
         self._check_inplace_setting(value)
         self._check_setitem_copy()
-        self._where(-key, value, inplace=True)
+        self._where(key, value, inplace=True, invert=True)
```

Comment:
so I would remove the invert keyword and just fix panda/core/series as well (and have _where *always* invert)

Reply:
Hmm, I think it could be confusing if the default behavior of `_where` is to do essentially the opposite of what it's done previously.

This keyword is only to handle the special case of inverting after filling when dealing with missing values / misalignment, and while it's not pretty it's at least not user-facing. I'm not really aware of an elegant way to do this given that realignment is already happening inside `_where`.

Reply:
_where is private so this doesn't matter, we want as consistent AND minimal api internally as possible.

Reply:
I can try to get this to work, but changing _where globally in this way breaks a large number of tests